internal/telemetry/client.go

```diff
@@ -92,6 +92,10 @@ type Client struct {
 		Printf(msg string, args ...interface{})
 	}
 
+	// Client will be used for telemetry uploads. If Client is nil, a
+	// default value will be used.
```

Comment:
Can you describe the behavior/side effect of using the default? I wouldn't know by reading this when it would be appropriate to rely on the default, and when I shouldn't.